ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -103,59 +105,67 @@ public IngestionResult ingestFromBlob(BlobSourceInfo blobSourceInfo, IngestionPr
-        } catch (IOException | URISyntaxException e) {
-            throw new IngestionClientException("Error in ingestFromBlob()", e);
+            throw new IngestionServiceException("Failed to ingest from blob", e);
+        } catch (IOException e) {
+            throw new IngestionClientException("Failed to ingest from blob", e);
         }
     }
 
     @Override
-    public IngestionResult ingestFromFile(FileSourceInfo fileSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
+    public IngestionResult ingestFromFile(FileSourceInfo fileSourceInfo, IngestionProperties ingestionProperties)
+            throws IngestionClientException, IngestionServiceException {
         // Argument validation:
-        if (fileSourceInfo == null || ingestionProperties == null) {
-            throw new IllegalArgumentException("fileSourceInfo or ingestionProperties is null");
-        }
+        validateIsNotNull(fileSourceInfo, "fileSourceInfo is null");
+        validateIsNotNull(ingestionProperties, "ingestionProperties is null");
+
         fileSourceInfo.validate();
         ingestionProperties.validate();
 
         try {
-            String fileName = (new File(fileSourceInfo.getFilePath())).getName();
-            String blobName = genBlobName(fileName, ingestionProperties.getDatabaseName(), ingestionProperties.getTableName());
-            CloudBlockBlob blob = azureStorageHelper.uploadLocalFileToBlob(fileSourceInfo.getFilePath(), blobName, resourceManager.getIngestionResource(ResourceManager.ResourceType.TEMP_STORAGE));
-            String blobPath = azureStorageHelper.getBlobPathWithSas(blob);
+            String filePath = fileSourceInfo.getFilePath();
+            File file = Validation.validateAndCreateFile(filePath);
+
+            String fileName = file.getName();
+            String blobName = genBlobName(
+                    fileName, ingestionProperties.getDatabaseName(), ingestionProperties.getTableName());
+            CloudBlockBlob blob = azureStorageClient.uploadLocalFileToBlob(fileSourceInfo.getFilePath(), blobName,
+                    resourceManager.getIngestionResource(ResourceManager.ResourceType.TEMP_STORAGE));
+            String blobPath = azureStorageClient.getBlobPathWithSas(blob);
             long rawDataSize = fileSourceInfo.getRawSizeInBytes() > 0L ? fileSourceInfo.getRawSizeInBytes() :
-                    estimateFileRawSize(fileSourceInfo.getFilePath());
+                    estimateFileRawSize(filePath);
 
             BlobSourceInfo blobSourceInfo = new BlobSourceInfo(blobPath, rawDataSize, fileSourceInfo.getSourceId());
 
             return ingestFromBlob(blobSourceInfo, ingestionProperties);
 
         } catch (StorageException e) {
-            throw new IngestionServiceException("Error in ingestFromFile()", e);
+            throw new IngestionServiceException("Failed to ingest from file", e);
```

Comment:
Why is this IngestionServiceException ?

Reply:
This is an Azure Storage exception. I think, for the user this is a Service exception, and not Client.